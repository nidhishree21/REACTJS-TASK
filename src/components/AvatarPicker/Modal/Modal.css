.Modal {
  position: relative;
  z-index: 500;
  width: 280px;
  margin-top: 10px;
  border-radius: 2px;
  background-color: rgb(44, 48, 51);
  box-shadow: 2px 2px 10px rgb(102, 102, 102);
  padding: 14px;
  outline: none;
}
.Modal::moz-focus-inner {
  border: 0;
}
.Modal::before {
  content: "";
  position: absolute;
  top: -16px;
  left: 50%;
  transform: translateX(-50%);
  z-index: 1;
  border: solid 8px transparent;
  border-bottom-color: rgb(44, 48, 51);
}

.DisplayModal {
  animation-duration: 0.2s;
  animation-name: bounceOver;
}

.HideModal {
  opacity: 0;
  transform: scale(0);
  transition: all 0.2s ease-out;
}

@keyframes bounceOver {
  from,
  40%,
  80%,
  to {
    animation-timing-function: cubic-bezier(0.5, 0.03, 0.15, 0.73);
  }

  0% {
    transform: scale3d(0.3, 0.3, 0.3);
  }

  40% {
    opacity: 1;
    transform: scale3d(0.7, 0.7, 0.7);
  }

  80% {
    transform: scale3d(1.2, 1.2, 1.2);
  }

  to {
    opacity: 1;
    transform: scale3d(1, 1, 1);
  }
}
