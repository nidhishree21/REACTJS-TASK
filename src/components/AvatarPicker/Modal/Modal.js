import React, { Component } from "react";
import PropTypes from "prop-types";

import Backdrop from "../Backdrop/Backdrop";
import styles from "./Modal.css";

class Modal extends Component {
  static propTypes = {
    show: PropTypes.bool.isRequired,
    hide: PropTypes.func.isRequired
  };

  componentDidMount() {
    window.addEventListener("keyup", this.props.hide, false);
  }
  componentWillUnmount() {
    window.removeEventListener("keyup", this.props.hide, false);
  }
  render() {
    return (
      <React.Fragment>
        <Backdrop show={this.props.show} hide={this.props.hide} />
        <div
          tabIndex="0"
          className={`${styles.Modal} ${
            this.props.show ? styles.DisplayModal : styles.HideModal
          }`}
        >
          {this.props.children}
        </div>
      </React.Fragment>
    );
  }
}

export default Modal;
