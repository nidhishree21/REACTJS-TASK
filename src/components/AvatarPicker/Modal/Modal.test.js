import React from "react";
import { shallow } from "enzyme";
import Modal from "./Modal";

const testProps = {
  show: true,
  hide: jest.fn(() => {})
};

describe("Modal", () => {
  it("renders Modal correctly", () => {
    const wrapper = shallow(<Modal {...testProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
