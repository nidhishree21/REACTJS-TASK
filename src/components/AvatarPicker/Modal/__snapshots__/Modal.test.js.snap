// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders Modal correctly 1`] = `
<React.Fragment>
  <Backdrop
    hide={[Function]}
    show={true}
  />
  <div
    className="Modal DisplayModal"
    tabIndex="0"
  />
</React.Fragment>
`;
